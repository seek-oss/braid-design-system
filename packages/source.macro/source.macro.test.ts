import pluginTester from 'babel-plugin-tester';
import plugin from 'babel-plugin-macros';

pluginTester({
  plugin,
  pluginName: 'source.macro',
  snapshot: true,
  babelOptions: {
    filename: 'dummy.tsx',
    plugins: [
      require.resolve('@babel/plugin-syntax-jsx'),
      [require.resolve('@babel/plugin-syntax-typescript'), { isTSX: true }],
    ],
  },
  tests: {
    'smoke test': {
      code: /* ts */ `
        import source from './source.macro';

        const result = source({
          // comment
          hello: () => 'world'
        });
      `,
    },
    'react component': {
      code: /* ts */ `
        import source from './source.macro';

        const result = source(
          <div />
        );
      `,
    },
    'ignores types': {
      code: /* ts */ `
        import type { Source } from './source.macro';

        const foo = 'bar';
      `,
    },
    'with new lines': {
      code: /* ts */ `
        import source from './source.macro';

        const options = source(() => {
          sideEffect();

          return [
            { value: "1", label: "Option 1" },
            { value: "2", label: "Option 2" },
            { value: "3", label: "Option 3" },
          ];
        });
      `,
    },
    'code only': {
      pluginOptions: { source: { codeOnly: true } },
      code: /* ts */ `
        import source from './source.macro';

        const result = source({
          hello: () => 'world',
          some: 'value',
        });
      `,
    },
  },
});
