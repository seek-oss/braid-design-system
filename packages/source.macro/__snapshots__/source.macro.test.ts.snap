// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`source.macro 1. smoke test: 1. smoke test 1`] = `


import source from './source.macro';

const result = source({
  // comment
  hello: () => 'world'
});
      

      ↓ ↓ ↓ ↓ ↓ ↓

const result = {
  code: "{\\n  // comment\\n  hello: () => 'world'\\n}",
  value: {
    // comment
    hello: () => 'world',
  },
};

`;

exports[`source.macro 2. react component: 2. react component 1`] = `


import source from './source.macro';

const result = source(
  <div />
);
      

      ↓ ↓ ↓ ↓ ↓ ↓

const result = {
  code: '<div />',
  value: <div />,
};

`;

exports[`source.macro 3. ignores types: 3. ignores types 1`] = `


import type { Source } from './source.macro';

const foo = 'bar';
      

      ↓ ↓ ↓ ↓ ↓ ↓

const foo = 'bar';

`;

exports[`source.macro 4. code only: 4. code only 1`] = `


import source from './source.macro';

const result = source({
  hello: () => 'world',
  some: 'value',
});
      

      ↓ ↓ ↓ ↓ ↓ ↓

const result = "{\\n  hello: () => 'world',\\n  some: 'value'\\n}";

`;
